src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/ExceptionRecordTest.java

```diff
@@ -31,11 +30,12 @@ public void should_dispatch_blob_and_create_exception_record_for_classification(
     }
 
     private void assertCompletedProcessorResult(String zipFileName) {
-        ProcessorEnvelopeResult processorEnvelopeResult = getZipFileStatus(zipFileName);
-        assertThat(processorEnvelopeResult.ccdId).isNotBlank();
-        assertThat(processorEnvelopeResult.container).isEqualTo(Container.BULKSCAN.name);
-        assertThat(processorEnvelopeResult.envelopeCcdAction).isEqualTo("EXCEPTION_RECORD");
-        assertThat(processorEnvelopeResult.id).isNotBlank();
-        assertThat(processorEnvelopeResult.status).isEqualTo("COMPLETED");
+        assertThat(getZipFileStatus(zipFileName)).hasValueSatisfying(env -> {
+            assertThat(env.ccdId).isNotBlank();
+            assertThat(env.container).isEqualTo(Container.BULKSCAN.name);
+            assertThat(env.envelopeCcdAction).isEqualTo("EXCEPTION_RECORD");
+            assertThat(env.id).isNotBlank();
+            assertThat(env.status).isEqualTo("COMPLETED");
+        });
```

Comment:
While I'm a fan of splitting codes into methods eagerly, in this case I'd probably bring it back to the test method, because these assumptions rely on some context and this method can't be easily reused, unless a few parameters are added to it.